test/studies/hpcc/STREAMS/elliot/stream-spmd-barrier.chpl

```diff
@@ -35,11 +38,21 @@ proc main() {
 
   var execTime: [1..numTrials] real;
 
-  var barrier = new Barrier(numTasks, barrierType);
   for trial in 1..numTrials {
+    var count: atomic int = numTasks;
+    var sgate: sync int;
+    var agate: atomic int;
+
     const startTime = timeSinceEpoch().totalSeconds();
     coforall tid in 0..#numTasks {
-      barrier.barrier();
+      // do a basic hand-coded barrier using either syncs or atomics
+      var c = count.fetchSub(1);
+      if c == 1 {
+        if barrierType == BarrierType.Sync then sgate.writeXF(1);
+                                           else agate.write(1);
+      }
+      if barrierType == BarrierType.Sync then sgate.readFF();
+                                         else agate.waitFor(1);
```

Comment:
TY -- I have verified this test still checks that syncs aren't serializing perf (checked that perf before and after your change is the same and that if I revert 48850d2db7ea95008320dc62d70f00127ef6feb2 the sync case gets bad perf again)